### PR TITLE
[DOC release] per #11893 warn when using a route named 'select' as may not work …

### PR DESCRIPTION
**Pull request by EricSchank**
…quite as expected

### CODE REVIEW

File: packages/ember-routing/tests/system/dsl_test.js [hunk 1]
```diff
@@ -1,13 +1,25 @@
 import isEnabled from 'ember-metal/features';
+import Ember from 'ember-metal/core';
 import EmberRouter from 'ember-routing/system/router';
 
-var Router;
+var Router, oldWarn;
+
+function confirmWarns(expectedMsg) {
```
Comment (mixonic) [inline packages/ember-routing/tests/system/dsl_test.js]: There is no general helper for `expectWarn`? I would not be surprised, but we do have them for asserts and deprecations so it is worth checking.

**Comment by mixonic** [comment]
Also, tests are failing here FYI!

**Comment by EricSchank** [comment]
Doh!
@mixonic I didn't see an `expectWarn`, but I'll look again.  Thanks!

**Comment by EricSchank** [comment]
Only test failing is the IE10 Sauce test, but doesn't seem to provide any useful output.

@mixonic Any suggestions on this one?  How do you generally tackle this kind of failure?  (I've used Sauce alternatives, but haven't used Sauce before -- so not sure how to access logs, etc.)

Thanks!

image

**Comment by rwjblue** [comment]
Restarted sauce labs tests.

**Comment by EricSchank** [comment]
Thanks @rwjblue !
Looks like they passed this time.

**Comment by EricSchank** [comment]
@rwjblue or @mixonic 
Would one of you guys mind merging?  Or do you see other changes that I should make on this?
Thanks!

**Comment by mixonic** [comment]
@EricSchank it looks like your test is gone?

**Comment by EricSchank** [comment]
@mixonic or @rwjblue 
I could use some advice here.
The tests pass on my system, but fail on Travis CI. Here's the main error I'm seeing on Travis CI:

```
Test failed: Ember Router DSL: should warn when using a dangerous route name
    Failed assertion: Expected 1 assertions, but 0 were run
    at http://localhost:13141/qunit/qunit.js:251
    at http://localhost:13141/ember-tests.prod.js:29287
    at a (http://localhost:13141/ember.min.js:1)
    at http://localhost:13141/ember.min.js:1
    at http://localhost:13141/tests/?skipPackage=container,ember-testing,ember-debug&dist=min&prod=true:164
```

Which I interpret as it not even hitting the assertion, but on my machine I see one more passing assertion with this code than I do without these changes -- so I believe it's hitting the assertion locally.
Any suggestions on how to recreate on my machine, or guidance from seeing this type of error before?

I really don't see many tests around `Ember.warn` calls, so I may have just approached this test in the wrong way.

Thanks!

**Comment by rwjblue** [comment]
Just chatted with @EricSchank in slack about this, I think we came up with a path forward (and identified some things for addition to contributing.md).

**Comment by rwjblue** [comment]
Finally green!  Thanks for working so hard on this one @EricSchank!

**Pull request merged**